### ISSUE TITLE
Error trying to read XDG_RUNTIME_DIR on Linux

### ISSUE THREAD

**Issue by fmbarina** [Notarin/hayabusa#23]
Output of `systemctl status hayabusa.service`:

```
jul 17 17:10:09 fedora systemd[1]: Started hayabusa.service - The hayabusa daemon.
jul 17 17:10:10 fedora hayabusa[1965]: Running as daemon
jul 17 17:10:14 fedora hayabusa[1965]: error: XDG_RUNTIME_DIR is invalid or not set in the environment.
jul 17 17:10:14 fedora hayabusa[1965]: error: XDG_RUNTIME_DIR is invalid or not set in the environment.
```

Output of `echo $XDG_RUNTIME_DIR`:
```
/run/user/1000
```

Running:
- Fedora Linux 40
- Hayabusa version 0.3.1

I don't understand this very well yet, but I'm guessing the hayabusa daemon can't know XDG_RUNTIME_DIR since it's not running under a user process?

From the [Arch Wiki](https://wiki.archlinux.org/title/XDG_Base_Directory):

>XDG_RUNTIME_DIR is set by default through pam_systemd(8).

From the [manual on pam_systemd](https://man.archlinux.org/man/pam_systemd.8):

>$XDG_RUNTIME_DIR is not set if the current user is not the original user of the session.

From the freedesktop.org [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):

>$XDG_RUNTIME_DIR defines the base directory relative to which user-specific non-essential runtime files and other file objects (such as sockets, named pipes, ...) should be stored. The directory MUST be owned by the user, and he MUST be the only one having read and write access to it. Its Unix access mode MUST be 0700.

Assuming I'm not severily misunderstanding these, doesn't this mean XDG_RUNTIME_DIR is not defined for the daemon, much like other user environment variables are scoped to the current user and generally not available outside of it? Not sure what to do about this, though.

**Comment by Notarin** [comment]
This is interesting, because I do not require the variable in the code I've written, which means some library is panicking, I'll have to figure out which.
Thank you of informing me.

**Comment by fmbarina** [comment]
Yeah, I later realized hayabusa doesn't use it directly, so I'm curious about the source of the error.

By the way, if you don't mind answering this, what's the rationale behind running a system service rather than a user one? Asking because, from my point of view, other fetch software can display similar information while running under the user, so it shouldn't be a matter of permissions. But I couldn't think of other reasons to do so.

**Comment by Notarin** [comment]
The reason for it running as a system service is because it is a system-wide service, its designed so that other software may fetch the same system info, other users may fetch system info, I see no reason to have the exact same service running multiple times being one for each user who uses a tool that requires hayabusa.

I should clarify, you *can* run it as a user service if you wish (I'm pretty sure, I don't recall testing it), there is a reason I provide a stand alone binary and a standalone service file.
